test/test_basic_logic.py

```diff
@@ -439,8 +439,16 @@ def test_dict_headers(self):
         """
         Sending headers using dict is deprecated but still valid.
         """
+        # One of the steps in the outbound header validation logic checks
+        # that we don't send a pseudo-header after we've sent a regular
+        # header.  We can't guarantee that if you send headers as a dict,
+        # because dicts are unordered.
+        config = h2.config.H2Configuration(
+            validate_outbound_headers=False
+        )
+
```

Comment:
Given that `validate_outbound_headers` is currently defaulting to `True`, I think this represents a backward compatibility violation: dict-headers that previously "worked" now throw exceptions.

Now, I *think* that's acceptable here because we're tightening bugs, but it's worth us using some language about this in the changelog if you can.